### PR TITLE
Allow users to use /roomauth in private rooms with modjoin with permission

### PR DESCRIPTION
**Pull request by Morfent**
…rank

### CODE REVIEW

**Comment by Morfent** [comment]
Don't merge this, my logic for it's flawed

**Comment by Morfent** [comment]
Should be fixed now

File: commands.js [hunk 1]
```diff
@@ -1177,10 +1177,13 @@ exports.commands = {
 		if (cmd === 'roomauth1') userLookup = '\n\nTo look up auth for a user, use /userauth ' + target;
 		let targetRoom = room;
 		if (target) targetRoom = Rooms.search(target);
-		let unavailableRoom = targetRoom && (targetRoom !== room && (targetRoom.modjoin || targetRoom.staffRoom) && !user.can('makeroom'));
-		if (!targetRoom || unavailableRoom) return this.errorReply("The room '" + target + "' does not exist.");
+		if (!targetRoom) return this.errorReply("The room '" + target + "' does not exist.");
 		if (!targetRoom.auth) return this.sendReply("/roomauth - The room '" + (targetRoom.title ? targetRoom.title : target) + "' isn't designed for per-room moderation and therefore has no auth list." + userLookup);
 
+		let cannotJoin = targetRoom.isPrivate && targetRoom.modjoin && !targetRoom.auth[user.userid];
+		let unavailableRoom = (targetRoom !== room) && !user.inRooms.has(targetRoom.id) && (cannotJoin || targetRoom.staffRoom) && !user.can('makeroom');
```
Comment (Zarel) [inline commands.js]: `(targetRoom !== room) && !user.inRooms.has(targetRoom.id)` simplifies to `!user.inRooms.has(targetRoom.id)`

**Comment by Morfent** [comment]
Fixed

**Comment by Morfent** [comment]
And made a very minor optimization

File: commands.js [hunk 2]
```diff
@@ -1177,9 +1177,12 @@ exports.commands = {
 		if (cmd === 'roomauth1') userLookup = '\n\nTo look up auth for a user, use /userauth ' + target;
 		let targetRoom = room;
 		if (target) targetRoom = Rooms.search(target);
-		let unavailableRoom = targetRoom && (targetRoom !== room && (targetRoom.modjoin || targetRoom.staffRoom) && !user.can('makeroom'));
-		if (!targetRoom || unavailableRoom) return this.errorReply("The room '" + target + "' does not exist.");
-		if (!targetRoom.auth) return this.sendReply("/roomauth - The room '" + (targetRoom.title ? targetRoom.title : target) + "' isn't designed for per-room moderation and therefore has no auth list." + userLookup);
+		if (!targetRoom) return this.errorReply("The room '" + target + "' does not exist.");
+		if (!targetRoom.auth) return this.sendReply("/roomauth - The room '" + (targetRoom.title || target) + "' isn't designed for per-room moderation and therefore has no auth list." + userLookup);
+
+		let cannotJoin = targetRoom.isPrivate && targetRoom.modjoin && !targetRoom.auth[user.userid];
```
Comment (Zarel) [inline commands.js]: `(cannotJoin || targetRoom.staffRoom) && !user.can('makeroom')` is the _real_ `cannotJoin`

**Comment by Morfent** [comment]
Fixed again.

File: commands.js (hunk 2 above)
Comment (Morfent) [inline commands.js]: Fair enough (message didn't get sent earlier)

**Comment by bumbadadabum** [comment]
This allows people to check roomauth in every modjoined room it seems like.

**Comment by bumbadadabum** [comment]
Yeah @Morfent I'm pretty sure it's room.modjoin and not room.modJoin.

**Comment by Morfent** [comment]
It is. Stupid mistake on my part from never being certain what variable/method/property names decide not to use camelcase for whatever reason, then forgetting to check if `room.modjoin` was one of those beforehand.

**Pull request merged**